pkg/requirements/requirements.go

```diff
@@ -91,6 +91,46 @@ func CloneClusterRepo(g gitclient.Interface, gitURL string) (string, error) {
 	return dir, nil
 }
 
+func CloneClusterRepoSparse(g gitclient.Interface, gitURL string, cloneType string, sparseCheckoutPatterns []string) (string, error) {
+	secretMountPath := os.Getenv(credentialhelper.GIT_SECRET_MOUNT_PATH)
+	if secretMountPath != "" {
+		err := credentialhelper.WriteGitCredentialFromSecretMount()
+		if err != nil {
+			return "", fmt.Errorf("failed to write git credentials file for secret %s : %w", secretMountPath, err)
+		}
+
+		gitURL, err = AddUserPasswordToURLFromDir(gitURL, secretMountPath)
+		if err != nil {
+			return "", fmt.Errorf("failed to add username and password to git URL: %w", err)
+		}
+	} else {
+		if kube.IsInCluster() {
+			log.Logger().Warnf("no $GIT_SECRET_MOUNT_PATH environment variable set")
+		} else {
+			log.Logger().Debugf("no $GIT_SECRET_MOUNT_PATH environment variable set")
+		}
+	}
+
+	// Decide on which clone method to use based on the cloneType flag
+	switch cloneType {
+	case "full":
+		dir, err := gitclient.CloneToDir(g, gitURL, "")
+		if err != nil {
+			return "", fmt.Errorf("failed to clone cluster git repo %s: %w", gitURL, err)
+		}
+		return dir, nil
```

Comment:
I don't think this case is neccasary anymore as `CloneClusterRepo()` exists

Reply:
Done